src/main/java/org/opensearch/security/configuration/SecurityFlsDlsIndexSearcherWrapper.java

```diff
@@ -38,24 +40,9 @@
 
 public class SecurityFlsDlsIndexSearcherWrapper extends SecurityIndexSearcherWrapper {
 
-    // TODO: the list is outdated. It is necessary to change how meta fields are handled in the near future.
-    // We may consider using MapperService.isMetadataField() instead of relying on the static set or
-    // (if it is too costly or does not meet requirements) use IndicesModule.getBuiltInMetadataFields()
-    // for OpenSearch version specific Set of meta fields
-    private static final Set<String> metaFields = Sets.newHashSet(
-        "_source",
-        "_version",
-        "_field_names",
-        "_seq_no",
-        "_primary_term",
-        "_id",
-        IgnoredFieldMapper.NAME,
-        "_index",
-        "_routing",
-        "_size",
-        "_timestamp",
-        "_ttl",
-        "_type"
+    private final Set<String> metaFields;
+    public static final Set<String> META_FIELDS_BEFORE_7DOT8 = Collections.unmodifiableSet(
+        new HashSet<>(Arrays.asList("_size", "_timestamp", "_ttl", "_type"))
```

Comment:
@peternied Quick update: "_ttl" and "_timestamp" seem to have been removed in ES 6.

The `TypeFieldMapper` was removed in OS 2.0: https://github.com/opensearch-project/OpenSearch/pull/3196

The "_size" FieldMapper comes from `mapper-size` plugin: https://github.com/opensearch-project/OpenSearch/blob/main/plugins/mapper-size/src/main/java/org/opensearch/index/mapper/size/SizeFieldMapper.java

I'm not sure if that should be considered built-in and in this includesSet by default.

Reply:
@cwperks Thanks for looking into this and charting out the history of these fields.

If `indexService.mapperService().getMetadataFields()` is getting metadata fields from the current index then it seems like the source of truth and we don't need a backup list, can you look into this?

Reply:
Yes, looking into it now. To recap: this list of metaFields is used by the security plugin for FLS when a role has FLS Include rules (as opposed to excludes). The idea of this list, is that metaFields are included by default in addition to fields that a user explicitly allows when using FLS includes.

The fields returned from `getMetadataFields` are driven by [IndicesModule.getMetadataMappers](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/indices/IndicesModule.java#L212-L242) which includes the built-in metadata mappers + mapper plugins (such as the mapper-size plugin).

I am going to create a localDistro of OpenSearch, install the mapper-size plugin and verify that "_size" is returned from `indexService.mapperService().getMetadataFields()`.

If that's the case, then I feel comfortable flipping this from draft and explicitly removing `"_timestamp", "_ttl", "_type"` which are legacy mappers "_timestamp" and "_ttl" were removed in ES 6 and "_type" was changed to "_nested_path" in OS 2.

Reply:
The documentation on the mapper-size plugin is lacking. To setup the mapper-size plugin you can create a local distribution of core and use `./bin/opensearch-plugin install mapper-size`.

Alternatively, you can create a zip and install it using:

```
./gradlew :plugins:mapper-size:assemble
./bin/opensearch-plugin install file:$OPENSEARCH_HOME/plugins/mapper-size/build/distributions/mapper-size-3.0.0-SNAPSHOT.zip
```

After you startup a cluster, you can create an index with the size mapping enabled, index a document and query like this:

```
curl -XPUT http://localhost:9200/example_index -H "Content-Type: application/json" -d '{
  "mappings": {
    "_size": {
      "enabled": true
    },
    "properties": {
      "name": {
        "type": "text"
      },
      "age": {
        "type": "integer"
      }
    }
  }
}'

curl -XPOST http://localhost:9200/example_index/_doc -H "Content-Type: application/json" -d '{
  "name": "John Doe",
  "age": 30
}'

curl -XGET http://localhost:9200/example_index/_search -H "Content-Type: application/json" -d '{
  "query": {
    "match_all": {}
  },
  "stored_fields": ["_size", "_source"]
}'

Query Results:

{"took":2,"timed_out":false,"_shards":{"total":1,"successful":1,"skipped":0,"failed":0},"hits":{"total":{"value":1,"relation":"eq"},"max_score":1.0,"hits":[{"_index":"example_index","_id":"Pctw0I8BLto8I5f_NLKK","_score":1.0,"_size":37,"_source":{
  "name": "John Doe",
  "age": 30
}}]}}
```